[
    {
        "number": 4,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "manifest_path": "docs/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-9wx4-h78v-vm56",
            "cve_id": "CVE-2024-35195",
            "summary": "Requests `Session` object does not verify requests after making first request with verify=False",
            "description": "When making requests through a Requests `Session`, if the first request is made with `verify=False` to disable cert verification, all subsequent requests to the same origin will continue to ignore cert verification regardless of changes to the value of `verify`. This behavior will continue for the lifecycle of the connection in the connection pool.\n\n### Remediation\nAny of these options can be used to remediate the current issue, we highly recommend upgrading as the preferred mitigation.\n\n* Upgrade to `requests>=2.32.0`.\n* For `requests<2.32.0`, avoid setting `verify=False` for the first request to a host while using a Requests Session.\n* For `requests<2.32.0`, call `close()` on `Session` objects to clear existing connections if `verify=False` is used.\n\n### Related Links\n* https://github.com/psf/requests/pull/6655",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-9wx4-h78v-vm56",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-35195",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/psf/requests/security/advisories/GHSA-9wx4-h78v-vm56"
                },
                {
                    "url": "https://github.com/psf/requests/pull/6655"
                },
                {
                    "url": "https://github.com/psf/requests/commit/a58d7f2ffb4d00b46dca2d70a3932a0b37e22fac"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-35195"
                },
                {
                    "url": "https://github.com/advisories/GHSA-9wx4-h78v-vm56"
                }
            ],
            "published_at": "2024-05-20T20:15:00Z",
            "updated_at": "2024-05-20T22:07:29Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "requests"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 2.32.0",
                    "first_patched_version": {
                        "identifier": "2.32.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:H/PR:H/UI:R/S:U/C:H/I:H/A:N",
                "score": 5.6
            },
            "cwes": [
                {
                    "cwe_id": "CWE-670",
                    "name": "Always-Incorrect Control Flow Implementation"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 2.32.0",
            "first_patched_version": {
                "identifier": "2.32.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/cleanlab/dependabot/alerts/4",
        "html_url": "https://github.com/fcas/cleanlab/security/dependabot/4",
        "created_at": "2024-05-22T23:24:03Z",
        "updated_at": "2024-05-22T23:24:03Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 3,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "Jinja2"
            },
            "manifest_path": "docs/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-h75v-3vvj-5mfj",
            "cve_id": "CVE-2024-34064",
            "summary": "Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter",
            "description": "The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for the previous GHSA-h5c8-rqwp-cp95 CVE-2024-22195 only addressed spaces but not other characters.\n\nAccepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-h75v-3vvj-5mfj",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2024-34064",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj"
                },
                {
                    "url": "https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-34064"
                },
                {
                    "url": "https://github.com/advisories/GHSA-h75v-3vvj-5mfj"
                }
            ],
            "published_at": "2024-05-06T14:20:59Z",
            "updated_at": "2024-05-06T16:17:32Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "Jinja2"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 3.1.4",
                    "first_patched_version": {
                        "identifier": "3.1.4"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N",
                "score": 5.4
            },
            "cwes": [
                {
                    "cwe_id": "CWE-79",
                    "name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "Jinja2"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 3.1.4",
            "first_patched_version": {
                "identifier": "3.1.4"
            }
        },
        "url": "https://api.github.com/repos/fcas/cleanlab/dependabot/alerts/3",
        "html_url": "https://github.com/fcas/cleanlab/security/dependabot/3",
        "created_at": "2024-05-22T23:24:03Z",
        "updated_at": "2024-05-22T23:24:03Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 2,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "manifest_path": "docs/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-j8r2-6x86-q33q",
            "cve_id": "CVE-2023-32681",
            "summary": "Unintended leak of Proxy-Authorization header in requests",
            "description": "### Impact\n\nSince Requests v2.3.0, Requests has been vulnerable to potentially leaking `Proxy-Authorization` headers to destination servers, specifically during redirects to an HTTPS origin. This is a product of how `rebuild_proxies` is used to recompute and [reattach the `Proxy-Authorization` header](https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/sessions.py#L319-L328) to requests when redirected. Note this behavior has _only_ been observed to affect proxied requests when credentials are supplied in the URL user information component (e.g. `https://username:password@proxy:8080`).\n\n**Current vulnerable behavior(s):**\n\n1. HTTP → HTTPS: **leak**\n2. HTTPS → HTTP: **no leak**\n3. HTTPS → HTTPS: **leak**\n4. HTTP → HTTP: **no leak**\n\nFor HTTP connections sent through the proxy, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the `Proxy-Authorization` header must be sent in the CONNECT request as the proxy has no visibility into further tunneled requests. This results in Requests forwarding the header to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate those credentials.\n\nThe reason this currently works for HTTPS connections in Requests is the `Proxy-Authorization` header is also handled by urllib3 with our usage of the ProxyManager in adapters.py with [`proxy_manager_for`](https://github.com/psf/requests/blob/f2629e9e3c7ce3c3c8c025bcd8db551101cbc773/requests/adapters.py#L199-L235). This will compute the required proxy headers in `proxy_headers` and pass them to the Proxy Manager, avoiding attaching them directly to the Request object. This will be our preferred option going forward for default usage.\n\n### Patches\nStarting in Requests v2.31.0, Requests will no longer attach this header to redirects with an HTTPS destination. This should have no negative impacts on the default behavior of the library as the proxy credentials are already properly being handled by urllib3's ProxyManager.\n\nFor users with custom adapters, this _may_ be potentially breaking if you were already working around this behavior. The previous functionality of `rebuild_proxies` doesn't make sense in any case, so we would encourage any users impacted to migrate any handling of Proxy-Authorization directly into their custom adapter.\n\n### Workarounds\nFor users who are not able to update Requests immediately, there is one potential workaround.\n\nYou may disable redirects by setting `allow_redirects` to `False` on all calls through Requests top-level APIs. Note that if you're currently relying on redirect behaviors, you will need to capture the 3xx response codes and ensure a new request is made to the redirect destination.\n```\nimport requests\nr = requests.get('http://github.com/', allow_redirects=False)\n```\n\n### Credits\n\nThis vulnerability was discovered and disclosed by the following individuals.\n\nDennis Brinkrolf, Haxolot (https://haxolot.com/)\nTobias Funke, (tobiasfunke93@gmail.com)",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-j8r2-6x86-q33q",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-32681",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/psf/requests/security/advisories/GHSA-j8r2-6x86-q33q"
                },
                {
                    "url": "https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-32681"
                },
                {
                    "url": "https://github.com/psf/requests/releases/tag/v2.31.0"
                },
                {
                    "url": "https://github.com/pypa/advisory-database/tree/main/vulns/requests/PYSEC-2023-74.yaml"
                },
                {
                    "url": "https://lists.debian.org/debian-lts-announce/2023/06/msg00018.html"
                },
                {
                    "url": "https://security.gentoo.org/glsa/202309-08"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AW7HNFGYP44RT3DUDQXG2QT3OEV2PJ7Y"
                },
                {
                    "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KOYASTZDGQG2BWLSNBPL3TQRL2G7QYNZ"
                },
                {
                    "url": "https://github.com/advisories/GHSA-j8r2-6x86-q33q"
                }
            ],
            "published_at": "2023-05-22T20:36:32Z",
            "updated_at": "2024-03-27T20:09:55Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "requests"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": ">= 2.3.0, < 2.31.0",
                    "first_patched_version": {
                        "identifier": "2.31.0"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:N/AC:H/PR:N/UI:R/S:C/C:H/I:N/A:N",
                "score": 6.1
            },
            "cwes": [
                {
                    "cwe_id": "CWE-200",
                    "name": "Exposure of Sensitive Information to an Unauthorized Actor"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "requests"
            },
            "severity": "medium",
            "vulnerable_version_range": ">= 2.3.0, < 2.31.0",
            "first_patched_version": {
                "identifier": "2.31.0"
            }
        },
        "url": "https://api.github.com/repos/fcas/cleanlab/dependabot/alerts/2",
        "html_url": "https://github.com/fcas/cleanlab/security/dependabot/2",
        "created_at": "2024-05-22T23:24:03Z",
        "updated_at": "2024-05-22T23:24:03Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    },
    {
        "number": 1,
        "state": "open",
        "dependency": {
            "package": {
                "ecosystem": "pip",
                "name": "IPython"
            },
            "manifest_path": "docs/requirements.txt",
            "scope": "runtime"
        },
        "security_advisory": {
            "ghsa_id": "GHSA-29gw-9793-fvw7",
            "cve_id": "CVE-2023-24816",
            "summary": "IPython vulnerable to command injection via set_term_title",
            "description": "IPython provides an interactive Python shell and Jupyter kernel to use Python interactively. Versions prior to 8.10.0 are vulnerable to command injection in the `set_term_title` [function](https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117) under specific conditions. This has been patched in version 8.10.0.\n\n### Impact \nUsers are only vulnerable when calling this function in Windows in a Python environment where [ctypes](https://docs.python.org/3/library/ctypes.html) is not available. The dependency on ctypes in `IPython.utils._process_win32` prevents the vulnerable code from ever being reached (making it effectively dead code). However, as a library that could be used by another tool, `set_term_title` could introduce a vulnerability for dependencies. Currently `set_term_title` is only called with (semi-)trusted input that contain the current working directory of the current IPython session. If an attacker can control directory names, and manage to get a user to `cd` into this directory, then the attacker can execute arbitrary commands contained in the folder names.",
            "severity": "medium",
            "identifiers": [
                {
                    "value": "GHSA-29gw-9793-fvw7",
                    "type": "GHSA"
                },
                {
                    "value": "CVE-2023-24816",
                    "type": "CVE"
                }
            ],
            "references": [
                {
                    "url": "https://github.com/ipython/ipython/security/advisories/GHSA-29gw-9793-fvw7"
                },
                {
                    "url": "https://github.com/ipython/ipython/commit/385d69325319a5972ee9b5983638e3617f21cb1f"
                },
                {
                    "url": "https://github.com/ipython/ipython/commit/991849c247fc208628879e7ca2923b3c218a5a75"
                },
                {
                    "url": "https://github.com/Carreau/ipython/blob/7557ade0ed927475d5ab5b573d0ea4febfb22683/docs/source/whatsnew/version8.rst#ipython-810"
                },
                {
                    "url": "https://nvd.nist.gov/vuln/detail/CVE-2023-24816"
                },
                {
                    "url": "https://github.com/ipython/ipython/blob/3f0bf05f072a91b2a3042d23ce250e5e906183fd/IPython/utils/terminal.py#L103-L117"
                },
                {
                    "url": "https://github.com/ipython/ipython/blob/56e6925dfa50e2c7f4a6471547b8176275db7c25/IPython/utils/_process_win32.py#L20"
                },
                {
                    "url": "https://github.com/advisories/GHSA-29gw-9793-fvw7"
                }
            ],
            "published_at": "2023-02-10T19:55:53Z",
            "updated_at": "2023-09-02T05:03:05Z",
            "withdrawn_at": null,
            "vulnerabilities": [
                {
                    "package": {
                        "ecosystem": "pip",
                        "name": "IPython"
                    },
                    "severity": "medium",
                    "vulnerable_version_range": "< 8.10",
                    "first_patched_version": {
                        "identifier": "8.10"
                    }
                }
            ],
            "cvss": {
                "vector_string": "CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:L/I:L/A:L",
                "score": 4.5
            },
            "cwes": [
                {
                    "cwe_id": "CWE-20",
                    "name": "Improper Input Validation"
                },
                {
                    "cwe_id": "CWE-78",
                    "name": "Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')"
                }
            ]
        },
        "security_vulnerability": {
            "package": {
                "ecosystem": "pip",
                "name": "IPython"
            },
            "severity": "medium",
            "vulnerable_version_range": "< 8.10",
            "first_patched_version": {
                "identifier": "8.10"
            }
        },
        "url": "https://api.github.com/repos/fcas/cleanlab/dependabot/alerts/1",
        "html_url": "https://github.com/fcas/cleanlab/security/dependabot/1",
        "created_at": "2024-05-22T23:24:02Z",
        "updated_at": "2024-05-22T23:24:02Z",
        "dismissed_at": null,
        "dismissed_by": null,
        "dismissed_reason": null,
        "dismissed_comment": null,
        "fixed_at": null,
        "auto_dismissed_at": null
    }
]